worlds/osrs/__init__.py

```diff
@@ -0,0 +1,640 @@
+
+            tasks_for_this_type = tasks_for_this_type[:max_amount_for_task_type]
+
+            tasks_per_task_type[task_type] = tasks_for_this_type
+            weights_per_task_type[task_type] = getattr(self.options,
+                                                       f"{task_type}_task_weight") if tasks_for_this_type else 0
+
+        # Build a list of collections and weights in a matching order for rnd.choices later
+        all_tasks = []
+        all_weights = []
+        for task_type in task_types:
+            all_tasks.append(tasks_per_task_type[task_type])
+            all_weights.append(weights_per_task_type[task_type])
+
+        while locations_added < locations_required or (generation_is_fake and len(all_tasks) > 0):
+            if all_tasks:
+                chosen_task = rnd.choices(all_tasks, all_weights)[0]
+                if chosen_task:
+                    task = chosen_task.pop()
+                    self.add_location(task)
+                    locations_added += 1
+
+                # This isn't an else because chosen_task can become empty in the process of resolving the above block
+                # We still want to clear this list out while we're doing that
+                if not chosen_task:
+                    index = all_tasks.index(chosen_task)
+                    del all_tasks[index]
+                    del all_weights[index]
+
+            else:
+                assert len(general_tasks) > 0, \
+                    "There are not enough available tasks to fill the remaining pool for OSRS. \
+                    Please adjust settings to be less restrictive of tasks."
+                task = general_tasks.pop()
+                self.add_location(task)
+                locations_added += 1
+                pass
+
+    def add_location(self, location):
+        index = [i for i in range(0, len(location_rows)) if location_rows[i].name == location.name][0]
+        self.create_and_add_location(index)
+
+    def create_items(self) -> None:
+        for item_row in item_rows:
+            if item_row.name != self.starting_area_item:
+                for c in range(item_row.count):
+                    item = self.create_item(item_row.name)
+                    self.multiworld.itempool.append(item)
+
+    def create_and_add_location(self, row_index) -> None:
+        location_row = location_rows[row_index]
+        # print(f"Adding task {location_row.name}")
+
+        # Create Location
+        location_id = self.base_id + row_index
+        if location_row.category == "points" or location_row.category == "goal":
+            location_id = None
+        location = OSRSLocation(self.player, location_row.name, location_id)
+        self.location_name_to_data[location_row.name] = location
+
+        # Add the location to its first region, or if it doesn't belong to one, to Menu
+        region = self.region_name_to_data["Menu"]
+        if len(location_row.regions) > 0:
```

Comment:
Could do this instead
```suggestion
        if location_row.regions:
```